scripts/west_commands/sign.py

```diff
@@ -506,9 +506,16 @@ def sign(self, command, build_dir, build_conf, formats):
 
         components = [ ] if (target in ('imx8', 'imx8m')) else [ bootloader ]
         components += [ kernel ]
-        sign_base += (args.tool_args +
-                     ['-o', out_bin] + conf_path_cmd + extra_ri_args +
-                     components)
+
+        sign_config_extra_args = config_get_words('rimage', 'extra-args', [])
+
+        if '-k' not in sign_config_extra_args + args.tool_args:
+            cmake_default_key = cache.get('RIMAGE_SIGN_KEY')
+            extra_ri_args += [ '-k', str(sof_src_dir / 'keys' / cmake_default_key) ]
+
+        # Warning: the _args order matters! Think hard.
```

Comment:
Surely a person who isn't familiar with this code would benefit from a better comment here :P